frontend/src/scenes/notebooks/Nodes/notebookNodeLogic.ts

```diff
@@ -233,17 +266,25 @@ export const notebookNodeLogic = kea<notebookNodeLogicType>([
                 props.updateAttributes({ __init: null })
             }
         },
+        unmount: () => {
+            props.notebookLogic.actions.unregisterNodeLogic(values.nodeId)
+        },
     })),
 
     afterMount(async (logic) => {
-        logic.props.notebookLogic.actions.registerNodeLogic(logic as any)
-        const resizeable = computeResizeable(logic.props.resizeable, logic.props.attributes)
-        logic.actions.setResizeable(resizeable)
-        logic.actions.initializeNode()
+        const { props, actions, values } = logic
+        props.notebookLogic.actions.registerNodeLogic(values.nodeId, logic as any)
+
+        const isResizeable =
+            typeof props.resizeable === 'function' ? props.resizeable(props.attributes) : props.resizeable ?? true
+
+        actions.setResizeable(isResizeable)
+        actions.initializeNode()
     }),
 
-    beforeUnmount((logic) => {
-        logic.props.notebookLogic.actions.unregisterNodeLogic(logic as any)
+    beforeUnmount(({ props, values }) => {
+        // Note this doesn't work as there may be other places where this is used. The NodeWrapper should be in charge of somehow unmounting this
```

Comment:
Could we do
```
const { unregisterNodeLogic } = useActions(notebookLogic)

useEffect(() => {
  return () => unregisterNodeLogic(values.nodeId)
}, [])
```

in the wrapper to achieve what this comment suggests?

Reply:
We could do that yeah.